jobs: create jobs ignoring user session timezone

dt opened the pull request:
Release note (bug fix): fixes a bug where jobs created in session with non-zero session timezone offsets could hang before starting or report incorrect creation times when viewed in SHOW JOBS and the console.

Epic: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/123632)
<!-- Reviewable:end -->

dt commented:
TODO test

stevendanna reviewed (commented):
👍 Definitely seems like one we want a test for as you mention, but the overall fix looks good.

dt committed:
jobs: create jobs ignoring user session timezone

Release note (bug fix): fixes a bug where jobs created in session with non-zero
session timezone offsets could hang before starting or report incorrect creation
times when viewed in SHOW JOBS and the console.

Epic: none.

dt commented:
TFTR!

bors r+
